shadow/tests/v2ClientTests.cpp

```diff
@@ -0,0 +1,624 @@
+}
+AWS_TEST_CASE(ShadowV2ClientGetNonExistentNamedShadow311, s_ShadowV2ClientGetNonExistentNamedShadow311)
+
+static void s_deleteNamedShadow(
+    std::shared_ptr<Aws::Iotshadow::IClientV2> shadowClient,
+    const Aws::Crt::String &thingName,
+    const Aws::Crt::String &shadowName)
+{
+    ResultWaiter<Aws::Iotshadow::DeleteNamedShadowResult> deleteResultWaiter;
+
+    Aws::Iotshadow::DeleteNamedShadowRequest deleteRequest;
+    deleteRequest.ShadowName = shadowName;
+    deleteRequest.ThingName = thingName;
+
+    shadowClient->DeleteNamedShadow(
+        deleteRequest,
+        [&deleteResultWaiter](Aws::Iotshadow::DeleteNamedShadowResult &&result)
+        { deleteResultWaiter.ApplyResult(std::move(result)); });
+
+    deleteResultWaiter.GetResult();
+}
+
+static int s_doCreateDestroyNamedShadowTest(
+    std::shared_ptr<Aws::Iotshadow::IClientV2> shadowClient,
+    const Aws::Crt::String &thingName,
+    const Aws::Crt::String &shadowName)
+{
+    if (!shadowClient)
+    {
+        return AWS_OP_SKIP;
+    }
+
+    // 1. Create the Shadow
+    ResultWaiter<Aws::Iotshadow::UpdateNamedShadowResult> updateResultWaiter;
+
+    Aws::Crt::JsonObject desired;
+    desired.WithString("color", "green");
+
+    Aws::Iotshadow::ShadowState state;
+    state.Desired = desired;
+
+    Aws::Iotshadow::UpdateNamedShadowRequest updateRequest;
+    updateRequest.ShadowName = shadowName;
+    updateRequest.ThingName = thingName;
+    updateRequest.State = state;
+
+    shadowClient->UpdateNamedShadow(
+        updateRequest,
+        [&updateResultWaiter](Aws::Iotshadow::UpdateNamedShadowResult &&result)
+        { updateResultWaiter.ApplyResult(std::move(result)); });
+
+    const auto &updateResult = updateResultWaiter.GetResult();
+    ASSERT_TRUE(updateResult.IsSuccess());
+
+    const auto &updateResponse = updateResult.GetResponse();
+    const auto &updateState = updateResponse.State.value();
+
+    auto desiredColor = updateState.Desired.value().View().GetString("color");
+    ASSERT_TRUE(desiredColor == "green");
+
+    // 2. Delete the shadow
+    ResultWaiter<Aws::Iotshadow::DeleteNamedShadowResult> deleteResultWaiter;
+
+    Aws::Iotshadow::DeleteNamedShadowRequest deleteRequest;
+    deleteRequest.ShadowName = shadowName;
+    deleteRequest.ThingName = thingName;
+
+    shadowClient->DeleteNamedShadow(
+        deleteRequest,
+        [&deleteResultWaiter](Aws::Iotshadow::DeleteNamedShadowResult &&result)
+        { deleteResultWaiter.ApplyResult(std::move(result)); });
+
+    const auto &deleteResult = deleteResultWaiter.GetResult();
+    ASSERT_TRUE(deleteResult.IsSuccess());
+
+    return AWS_OP_SUCCESS;
+}
+
+static int s_ShadowV2ClientCreateDestroyNamedShadow5(Aws::Crt::Allocator *allocator, void *)
+{
+    Aws::Crt::ApiHandle handle;
+    auto protocolClient = s_createProtocolClient5(allocator);
+    auto client = s_createShadowClient5(protocolClient, allocator);
+    auto thingName = Aws::Crt::UUID().ToString();
```

Comment:
This Thing should be explicitly created at some point, right? I think you mentioned this once, or maybe I'm confusing it with something else.

Reply:
Jobs needs a thing because jobs target things by ARN, but you can manipulate shadows for things that don't explicitly exist without issue.